security/secret-management.md

```diff
@@ -0,0 +1,108 @@
+mitigating the risks associated with secrets leaked in the codebase. This ensures that our systems remain secure and 
+that sensitive information is protected against unauthorized access.
+
+##  Scope
+
+_* This policy applies to all software development projects, repositories, and environments within VisitScotland. It 
+covers the detection, handling, and remediation of leaked secrets across all stages of the software development 
+lifecycle._
+
+## Definitions
+
+- **Secret**: Sensitive information such as passwords, API keys, tokens, encryption keys, and other credentials that 
+  grant access to systems or data or authenticate requests
+- **Production Secret**: A secret used in the production environment, essential for the operation of live systems. These
+  secrets are rarely shared with developers and they tend to be used only in production and staging environments.
+- **Test Secret**: A secret used in non-production environments, such as development or testing, used for testing 
+  purposes only. However, Test secrets need to be carefully handled as they might grant access to services or result in
+  a breach of agreement with third-party systems. 
+- **Non-Secret Key**: An API key that does not provide access to sensitive data or critical functionalities and is 
+  considered low-risk if exposed. The most common example is a key that needs to be used by JavaScript and can be read 
+  in the source code of a page. While it is not classified as a secret, it is still preferred to manage it properly to 
+  maintain best practices in code management and security hygiene. See more information about 
+  [Non-Secret Key Management](key-management.md)
+
+### Types of Secrets
+
+- **API Keys**: Credentials used to authenticate applications with external services.
+- **Passwords**: Credentials used to access databases, services, or systems.
+- **Tokens**: Access tokens or refresh tokens used for authentication and authorization.
+- **Certificates**: SSL/TLS certificates or other cryptographic keys used for securing communications.
+- **Environment Variables**: Sensitive data stored as environment variables in deployment configurations.
+
+
+## What to do when a vulnerability is found
+
+### Assessment
+
+1. **Alert**: Use GitGuardian to detect and alert on leaked secrets on code commit.  
+2. **Review**: If the secret is a false positive. In case of false positive resolve the issue and including an 
+   explanation why the issue is considered a false positive. Note: Test Secrets are not false positive and they need to
+   be carefully considered as they might grant access to public facing services
+3. **Determine the Type**: Identify whether the leaked secret is a production secret, a test secret or a non-secret key.
+4. **Assess the Impact**: Evaluate the severity and potential impact of the leak on systems, data, and user security.
+5. **Open Source**: Identify whether the project is Open Source or not. Open Source projects
```

Comment:
This sentence needs completing.

Do we need to identify if it's Open Source or not? Does it affect the process?

Reply:
@James, yes this was an uncompleted thought I started to extend. 

An Open-source project needs to address Critical vulnerabilities quicker than a closed source project because we are essentially disclosing that we are vulnerable.  

My though behind this is that it is easier to target attacks to services searching for vulnerable dependency versions than just running services.

Reply:
I have also completed the sentence :)